airbyte-cdk/python/unit_tests/sources/file_based/scenarios/scenario_builder.py

```diff
@@ -59,26 +44,17 @@ def __init__(
         self.expected_check_error = expected_check_error
         self.expected_discover_error = expected_discover_error
         self.expected_read_error = expected_read_error
-        self.source = InMemoryFilesSource(
-            files,
-            file_type,
-            availability_strategy,
-            discovery_policy,
-            validation_policies,
-            parsers,
-            stream_reader,
-            self.configured_catalog(SyncMode.incremental if incremental_scenario_config else SyncMode.full_refresh),
-            file_write_options,
-            cursor_cls,
-        )
         self.incremental_scenario_config = incremental_scenario_config
         self.validate()
 
     def validate(self) -> None:
         assert self.name
         if not self.expected_catalog:
             return
-        streams = {s["name"] for s in self.config["streams"]}
```

Comment:
config["streams"] is specific to file-based. An alternative approach is to use `source.streams()` with the caveat that we should only verify the streams exist when there are no configuration errors